### DISCUSSION TITLE
How can I use my own docker image in an action?

### DISCUSSION THREAD

**Discussion by tombrus** [community/community#26417] [Actions]
<p>I recently discovered how to publish my own docker image to <em><a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a></em>.<br>
I thought it would be handy so I can use a custom configured image in my actions. That would save me precious time in every action-run for installing all kinds of utils I need.</p>
<p>How can I set the <strong>runs-on</strong> of a job to something so that it would use my custom image from <em><a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a></em>?</p>
<p>Is this possible at all or am I misunderstanding something here?</p>
<p>I can imagine that it can only work if I base my custom image on one of the prepared GitHub <strong>runs-on</strong> images.</p>
<hr>
<p>For those who are looking into their area:</p>
<blockquote>
<p>I made an example at <a href="https://github.com/tombrus/example-docker-image-publish" rel="noopener nofollow ugc">https://github.com/tombrus/example-docker-image-publish</a>. It publishes an image and uses it in an action job. Check it out.</p>
</blockquote>

**Comment by airtower-luna** [comment, marked as answer]
<p>You can do that with <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer"><code>jobs.&lt;job_id&gt;.container</code></a>, something like:</p>
<pre><code class="lang-yaml">jobs:
  build:
    runs-on: ubuntu-latest
    container:
      image: ghcr.io/your/image:latest
</code></pre>

**Comment by tombrus** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="141037">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<pre><code class="lang-auto">jobs:
  build:
    runs-on: ubuntu-latest
    container:
      image: ghcr.io/your/image:latest
</code></pre>
</blockquote>
</aside>
<p>Great, this works! Thanks.</p>
<p>For those who are looking into their area:</p>
<blockquote>
<p>I made an example at <a href="https://github.com/tombrus/example-docker-image-publish" rel="noopener nofollow ugc">https://github.com/tombrus/example-docker-image-publish</a>. It publishes an image and uses it in an action job. Check it out.</p>
</blockquote>
<p>I took some inspiration from your <a href="https://github.com/airtower-luna/hello-ghcr" rel="noopener nofollow ugc">https://github.com/airtower-luna/hello-ghcr</a> repo <a class="mention" href="/airtower-luna">@airtower-luna</a>, thanks!</p>
